lib/tasks/create_customer_support_download.rake

```diff
@@ -0,0 +1,50 @@
+task create_customer_support_download: :environment do
+  ### This is a utility that serves as a kind of "customer support dropbox".
+  ### The task will create a BulkDownload record that the user can use to download a file you wish to transfer to them.
+  ### The task will print instructions on the S3 location where you should put the file you have created for the user.
+  ### Usage: DESTINATION_USER_ID=<int> rake create_customer_support_download
+
+  user_id = ENV['DESTINATION_USER_ID'].to_i
+
+  user = User.find(user_id)
+  is_user_correct = await_confirmation("Is #{user.name} (#{user.email}) the user you wish to send your file to?",
+                                       "Please verify your DESTINATION_USER_ID and start over")
+  exit 2 unless is_user_correct
+
+  puts "Please enter a user-facing description of the file you wish to transfer:"
+  description = ActionController::Base.helpers.strip_tags(STDIN.gets.strip)
+
+  bulk_download = BulkDownload.create(download_type: "customer_support_request",
+                                      status: BulkDownload::STATUS_WAITING,
+                                      description: description,
+                                      user_id: user.id)
+  puts "BulkDownload record of type customer_support_request has been created."
+  s3_uri = "s3://#{ENV['SAMPLES_BUCKET_NAME']}/#{bulk_download.download_output_key}"
+  puts "Please upload the file you wish to transfer to '#{s3_uri}' using a command like:"
+  puts "\n  aws s3 cp <your file> '#{s3_uri}'"
+  puts "\n(Note the quotes around the destination path to take care of spaces in the destination file name.)"
+  is_file_uploaded = await_confirmation("Have you uploaded your file? Please confirm once you're ready",
+                                        "Process aborted, deleting the draft BulkDownload")
+  unless is_file_uploaded
+    bulk_download.destroy
+    exit 2
+  end
+
+  begin
+    bulk_download.mark_success
```

Comment:
I would rename `mark_success` to `verify_and_mark_success` and add a comment in that function that `fetch_output_file_size` will raise an error if the file isn't found. Otherwise, the code is quite confusing to read.

If you're open to paying down a bit of technical debt, it would be great to move these lines from `generate_download_file` into `mark_success`:

```
  rescue
    update(status: STATUS_ERROR)
    raise
  end
```

and renaming `mark_success` to `finalize`. It's a little weird that `mark_success` can raise an error and it expects the caller to handle it.

Reply:
This is a great point. I have renamed the method to `verify_and_mark_success` and added the comment next to `fetch_output_file_size`.

I have not made the change of moving the rescue block into the `mark_success` method because this would also mean that errors in the main block of `generate_download_file` are no longer caught, with potential side effects I cannot adequately assess. Prefer to leave a change of that logic to someone who is more intimately familiar with that code!